lib/vanagon/platform/windows.rb

```diff
@@ -53,9 +53,36 @@ def generate_packaging_artifacts(workdir, name, binding)
       # @param name [String] name of the project
       # @param binding [Binding] binding to use in evaluating the packaging templates
       def generate_msi_packaging_artifacts(workdir, name, binding)
-        FileUtils.mkdir_p(File.join(workdir, "wix"))
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.wxs.erb"), File.join(workdir, "wix",  "#{name}.wxs"), false, { :binding => binding })
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.filter.xslt.erb"), File.join(workdir, "wix", "#{name}.filter.xslt"), false, { :binding => binding })
+        # Recurse through the list of files in the resources/windows/wix directory and copy them over to the work directory.
+        copy_wix_artifacts_to_workdir("resources/windows/wix", workdir, name, binding)
+        copy_wix_artifacts_to_workdir(File.join(VANAGON_ROOT, "resources/windows/wix"), wordir, name, binding)
+      end
+
+      # Method to copy files over to the wix working directory from both the product
+      # specific directories (e.g. puppet-agent) and the generic (vanagon)
+      #
+      # @param srcdir [String] Originating directory
+      # @param workdir [String] working directory to stage the evaluated templates in
+      # @param name [String] name of the project
+      # @param binding [Binding] binding to use in evaluating the packaging templates
+      def copy_wix_artifacts_to_workdir(srcdir, workdir, name, binding, specials)
+
+        currentwd = Dir.getwd
+        Dir.chdir(srcdir)
```

Comment:
Oh, that being said, is the `chdir` even needed, or could you just glob for the files with a longer path (`"%{srcdir}/**/*.*"`) and operate on the files found?